benchmarks/helper.py

```diff
@@ -0,0 +1,222 @@
+
+rcParams.update({"figure.autolayout": True})
+plt.rc("axes", axisbelow=True)
+
+DEFAULT_OUTPUT_DIR = "benchmark_logs"
+
+
+TABLE = {
+    "suites": ["torchbench", "huggingface"],
+    "dtypes": ["float32", "float16", "amp"],
+    "devices": ["cuda", "cpu"],
+    "modes": ["inference", "training"],
+    # Dict of name to base_command. nightly is a special field to tell which compilers are relevant.
+    "training": {
+        "ts_nnc": "--training --speedup-ts --use-eval-mode --isolate",
+        "ts_nvfuser": "--training --nvfuser --speedup-ts --use-eval-mode --isolate",
+        "aot_eager": "--training --accuracy-aot-nop --generate-aot-autograd-stats --use-eval-mode --isolate",
+        "aot_nnc": "--training --accuracy-aot-ts-mincut --use-eval-mode --isolate",
+        "aot_nvfuser": "--training --nvfuser --accuracy-aot-ts-mincut --use-eval-mode --isolate",
+        "nightly": ["ts_nvfuser", "aot_nvfuser"],
+    },
+    # Dict of name to base_command. nightly is a special field to tell which compilers are relevant.
+    "inference": {
+        "ts_nnc": "-dcuda --isolate --speedup-ts",
+        "ts_nvfuser": "-dcuda --isolate -n100 --speedup-ts --nvfuser",
+        "trt": "-dcuda --isolate -n100 --speedup-trt",
+        "eager_cudagraphs": "-dcuda --inductor-settings --float32 -n50 --backend=cudagraphs",
+        "nnc_cudagraphs": "-dcuda --inductor-settings --float32 -n50 --backend=cudagraphs_ts --nvfuser",
+        "nvfuser_cudagraphs": "-dcuda --inductor-settings --float32 -n50 --backend=cudagraphs_ts",
+        "inductor_cudagraphs": "-dcuda --inductor-settings --float32 -n50 --inductor",
+        "nightly": ["nvfuser_cudagraphs", "inductor_cudagraphs"],
+    },
+}
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--filter", "-k", action="append", help="filter benchmarks with regexp"
+    )
+    parser.add_argument("--devices", action="append", help="cpu or cuda")
+    parser.add_argument("--modes", action="append", help="inference or training")
+    parser.add_argument("--dtypes", action="append", help="float16/float32/amp")
+    parser.add_argument("--suites", action="append", help="huggingface/torchbench")
+    parser.add_argument("--quick", action="store_true", help="Just runs one model")
+    parser.add_argument(
+        "--nightly",
+        action="store_true",
+        help="Use only compilers mentioned in nightly field",
+    )
+    parser.add_argument(
+        "--output-dir", help="Choose the output directory to save the logs"
+    )
+
+    # Choose either generation of commands, pretty parsing or e2e runs
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument("--commands", action="store_true", help="Generate commands")
+    group.add_argument("--parse", action="store_true", help="Parse the files")
+    group.add_argument(
+        "--e2e", action="store_true", help="Generate commands, run and parse the files"
+    )
```

Comment:
@ngimel This is the mutually exclusive set of options.

1) Commands - Just generate the commands and save them in run.sh
2) Parse - Parse the already generated logs, and pretty print/graphs etc
3) e2e - (which in hindsight is a hideous name) generates command, runs the tests, and parses the logs as well.

Reply:
maybe make these changes?
  commands -> print_run_commands
  parse -> visualize_logs
  e2e -> run
  
and should 'run' be the default, such that 'required=False' for this group?